library/src/scala/tasty/Reflection.scala

```diff
@@ -2679,14 +2679,30 @@ trait Reflection { reflection =>
       /** Annotations attached to this symbol */
       def annots: List[Term]
 
+      /** Does this symbol come from a currently compiled source file? */
       def isDefinedInCurrentRun: Boolean
 
+      /** Dummy val symbol that owns all statements within the initialization of the class.
+       *  This may also contain local definitions such as classes defined in a `locally` block in the class.
+       */
       def isLocalDummy: Boolean
```

Comment:
Regarding `isLocalDummy` and `isRefinementClass`,  it would be better to not provide them if they are not useful in practice, as they involve implementation details that might change.

Reply:
We should see if we can have an alternative design. We cannot return the owner of the local dummy as that would be ambiguous.